meta-xt-control-domain-virtio/recipes-guest/domu/files/domu-ExecStart.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+DOMU_ID=""
+
+create_domain() {
+    echo "Creating DomU"
+    /usr/sbin/xl create /etc/xen/domu.cfg
+    echo "Created DomU"
+
+    # Fetch DomU Xen domain identifier
+    while [ "$DOMU_ID" == "" ]; do
+        DOMU_ID=$(xl list | awk '{ if ($1 == "DomU") print $2 }')
```

Comment:
Hi @lorc, In case of usage of the "xl domid" command, I'll need to additionally check for the error state (which happens when the domain is not available ) to avoid assigning the error content to the DOMD_ID variable.

The error looks like this:

root@generic-armv8-xt-dom0:~# xl domid non_existing_domain_id
Can't get domid of the domain name 'non_existing_domain_id', maybe this domain does not exist.
It'll take more lines of code and won't be that elegant.

awk, in turn, will return an empty string in case of a non-existing domain. That is more suitable in this specific case.

Reply:
It is as simple as `DOMU_ID=$(xl domid DomU 2>/dev/null)`. Really, not need to iterate over all domains and then pipe output to `awk` when you can use a dedicated command.

Reply:
Hi @lorc, now I got the idea. Thanks for the extensive explanation. Accepted.

Reply:
Hi @lorc, done.